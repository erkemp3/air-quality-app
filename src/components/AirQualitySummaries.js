import React from "react";
import PropTypes from "prop-types";
import AirQualitySummary from "./AirQualitySummary";

const AirQualitySummaries = ({ onSubmit, summaries }) => {
  const summaryKey = (name, id) => {
    return `${name}_${id}`;
  };

  return (
    <div className="summaries">
      {summaries.map((summary, i) => (
        <AirQualitySummary
          summary={summary}
          key={summaryKey(summary.location, i)}
          id={i}
          onSubmit={onSubmit}
        />
      ))}
    </div>
  );
};

AirQualitySummaries.propTypes = {
  summaries: PropTypes.arrayOf(
    PropTypes.shape({
      city: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      measurements: PropTypes.arrayOf(
        PropTypes.shape({
          lastUpdated: PropTypes.string.isRequired,
          parameter: PropTypes.string.isRequired,
          value: PropTypes.number.isRequired,
        }).isRequired
      ).isRequired,
    }).isRequired
  ).isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default AirQualitySummaries;
