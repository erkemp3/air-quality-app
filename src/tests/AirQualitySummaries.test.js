import React from "react";
import { render } from "@testing-library/react";
import AirQualitySummaries from "../components/AirQualitySummaries";

describe("AirQualitySummaries", () => {
  const validProps = {
    summaries: [
      {
        city: "Leeds",
        location: "Leeds Centre",

        measurements: [
          {
            lastUpdated: "2021-02-01",
            parameter: "pm10",
            value: 9,
          },
          {
            lastUpdated: "2021-03-01",
            parameter: "s02",
            value: 10,
          },
        ],
      },
      {
        city: "Manchester",
        location: "Manchester Centre",

        measurements: [
          {
            lastUpdated: "2021-04-01 19:00",
            parameter: "c02",
            value: 11,
          },
          {
            lastUpdated: "2021-05-01",
            parameter: "pm10",
            value: 15,
          },
        ],
      },
    ],
    onSubmit: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <AirQualitySummaries
        summaries={validProps.summaries}
        city={validProps.city}
        location={validProps.location}
        measurements={validProps.measurements}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(
      <AirQualitySummaries
        summaries={validProps.summaries}
        city={validProps.city}
        location={validProps.location}
        measurements={validProps.measurements}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(getByText("Leeds Centre")).toHaveClass("location");
    expect(getByText("in Manchester, United Kingdom")).toHaveClass("city");
  });
});
