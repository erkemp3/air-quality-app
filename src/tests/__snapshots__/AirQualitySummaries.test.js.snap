// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirQualitySummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="summaries"
  >
    <div
      class="summary"
    >
      <div
        class="button-div"
      >
        <button
          class="delete-button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <p
        class="date"
      >
        UPDATED
            8 MONTHS AGO
      </p>
      <p
        class="location"
      >
        Leeds Centre
      </p>
      <p
        class="city"
      >
        in Leeds, United Kingdom
      </p>
      <p
        class="values"
      >
        Values: PM10: 9, S02: 10
      </p>
    </div>
    <div
      class="summary"
    >
      <div
        class="button-div"
      >
        <button
          class="delete-button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <p
        class="date"
      >
        UPDATED
            6 MONTHS AGO
      </p>
      <p
        class="location"
      >
        Manchester Centre
      </p>
      <p
        class="city"
      >
        in Manchester, United Kingdom
      </p>
      <p
        class="values"
      >
        Values: C02: 11, PM10: 15
      </p>
    </div>
  </div>
</DocumentFragment>
`;
